src/devices/src/virtio/device.rs

```diff
@@ -105,4 +134,15 @@ pub trait VirtioDevice: AsAny + Send {
     fn reset(&mut self) -> Option<(EventFd, Vec<EventFd>)> {
         None
     }
+
+    fn state(&self) -> VirtioDeviceState {
```

Comment:
Not sure if we should include this in `trait VirtioDevice` or add it to a child trait, smth like:
```rust
trait VirtioDeviceSerialization : VirtioDeviceState { // needs a better name
    fn state(&self) -> VirtioDeviceState { ... }
}
```

I am personally inclined towards the latter so that we don't pollute the `VirtioDevice` trait for users that don't need serialization.

Reply:
How about `VirtioDeviceSave`?